base/ltxref.dtx

```diff
@@ -108,7 +108,8 @@
 %  incremented by the command  \refstepcounter{CNT} , which sets
 %  \@currentlabel == {CNT}{eval(\p@cnt\theCNT)}.   The command
 %  \label{FOO} then writes the following on file \@auxout :
-%        \newlabel{FOO}{{eval(\@currentlabel)}{eval(\thepage)}}
+%        \newlabel{FOO}{{eval(\@currentlabel)}{eval(\thepage)}%
+%           {eval(\@currentlabelname)}{eval(\@currentHref)}{eval(\@kernel@currentdata)}}
```

Comment:
\@kernel@currentdata  is a very strange name ... for starters it is not at all linked to label/ref from the name

Reply:
> @kernel@currentdata is a very strange name ... for starters it is not at all linked to label/ref from the name

well currently it doesn't hold anything. I didn't want to leave it empty so that one can test in latex-lab what is needed or useful here and added some name that reserves that for us. Do you a have idea?

Reply:
Maybe  \ref@kernel@data 
or   \ref@kernel@currentdata